### DISCUSSION TITLE
logout without redirect

### DISCUSSION THREAD

**Discussion by mind0bender** [sergiodxa/remix-auth#229] [Q&A]
Assume I have an authenticator named userAuthenticator.

I have code like:
```ts
userAuthenticator
        .logout(request, {
          redirectTo: "/",
        })
        .then((): void => {
          resolve({
            success: true,
            data: {
              content: `Logged out of ${user.name}`,
            },
          });
        })
        .catch((err: any): void => {
            resolve(ERR500());
        });
```

I don't want to redirect the user after logout.
but in this case as the response is "THROWN" it is caught in the catch block and my server sends a 500.

**Comment by sergiodxa** [comment]
You can implement the logout yourself by destroying the session and sending it in a response.

The reason why the logout function redirects is because in my experience that’s the most common response you do, and it’s also simple to throw it and ensure any dev using it don’t forget to return a response with the Set-Cookie header and destroying the session.

**Reply by mind0bender** [reply]
that makes sense,
but an overload would surely be useful for cases like mine to pass null/false.

Thanks for the fast reply, 😃.
